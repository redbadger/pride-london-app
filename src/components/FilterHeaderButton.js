// @flow
import React from "react";
import { StyleSheet, View } from "react-native";
import type { ViewStyleProp } from "react-native/Libraries/StyleSheet/StyleSheet";
import Text from "./Text";
import Touchable from "./Touchable";
import { filterButtonTextColor, whiteColor } from "../constants/colors";
import NumberBadge from "./NumberBadge";

type Props = {
  text: string,
  label: string,
  onPress: Function,
  onRef?: Function,
  style?: ViewStyleProp,
  badgeValue?: ?number,
  active: boolean
};

const FilterHeaderButton = ({
  text,
  onPress,
  onRef,
  style,
  badgeValue,
  active,
  label
}: Props) => (
  <Touchable
    style={[styles.button, style]}
    onPress={onPress}
    ref={onRef}
    accessibilityLabel={label}
  >
    <Text
      type="small"
      style={[
        styles.buttonText,
        active ? styles.activeColor : styles.inactiveColor
      ]}
    >
      {text}
    </Text>
    {badgeValue != null && (
      <View style={styles.badgeContainer}>
        <NumberBadge value={badgeValue} />
      </View>
    )}
  </Touchable>
);

FilterHeaderButton.defaultProps = {
  onRef: undefined,
  style: {},
  badgeValue: null
};

const styles = StyleSheet.create({
  button: {
    paddingHorizontal: 8,
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "row"
  },
  buttonText: {
    fontFamily: "Roboto-Medium",
    textAlign: "center"
  },
  activeColor: {
    color: filterButtonTextColor
  },
  inactiveColor: {
    color: whiteColor
  },
  badgeContainer: {
    marginLeft: 6
  }
});

export default FilterHeaderButton;
