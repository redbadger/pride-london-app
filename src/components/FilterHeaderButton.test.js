// @flow
import React from "react";
import { shallow } from "enzyme";
import FilterHeaderButton from "./FilterHeaderButton";

it("renders correctly", () => {
  const style = { marginTop: 16 };
  const output = shallow(
    <FilterHeaderButton
      active
      text="1-2-3 filtered"
      label="1-2-3 filtered"
      style={style}
      onPress={() => {}}
    />
  );
  expect(output).toMatchSnapshot();
});

it("renders badge value when provided", () => {
  const style = { marginTop: 16 };
  const output = shallow(
    <FilterHeaderButton
      active
      text="1-2-3 filtered"
      label="1-2-3 filtered"
      style={style}
      onPress={() => {}}
      badgeValue={2}
    />
  );
  expect(output).toMatchSnapshot();
});

it("renders correct styles when inactive", () => {
  const style = { marginTop: 16 };
  const output = shallow(
    <FilterHeaderButton
      active={false}
      text="1-2-3 filtered"
      label="One two three filtered"
      style={style}
      onPress={() => {}}
    />
  );
  expect(output).toMatchSnapshot();
});
