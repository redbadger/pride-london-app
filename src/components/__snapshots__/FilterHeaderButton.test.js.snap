// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders badge value when provided 1`] = `
<Touchable
  accessibilityComponentType="button"
  accessibilityLabel="1-2-3 filtered"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  delayPressIn={50}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 8,
      },
      Object {
        "marginTop": 16,
      },
    ]
  }
>
  <Text
    color="blackColor"
    markdown={false}
    markdownStyle={Object {}}
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Medium",
        },
        Object {
          "color": "rgb(44, 218, 157)",
        },
      ]
    }
    type="text"
  >
    1-2-3 filtered
  </Text>
  <Component
    style={
      Object {
        "marginLeft": 6,
      }
    }
  >
    <NumberBadge
      value={2}
    />
  </Component>
</Touchable>
`;

exports[`renders correct styles when inactive 1`] = `
<Touchable
  accessibilityComponentType="button"
  accessibilityLabel="One two three filtered"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  delayPressIn={50}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 8,
      },
      Object {
        "marginTop": 16,
      },
    ]
  }
>
  <Text
    color="blackColor"
    markdown={false}
    markdownStyle={Object {}}
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Medium",
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    type="text"
  >
    1-2-3 filtered
  </Text>
</Touchable>
`;

exports[`renders correctly 1`] = `
<Touchable
  accessibilityComponentType="button"
  accessibilityLabel="1-2-3 filtered"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  delayPressIn={50}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 8,
      },
      Object {
        "marginTop": 16,
      },
    ]
  }
>
  <Text
    color="blackColor"
    markdown={false}
    markdownStyle={Object {}}
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Medium",
        },
        Object {
          "color": "rgb(44, 218, 157)",
        },
      ]
    }
    type="text"
  >
    1-2-3 filtered
  </Text>
</Touchable>
`;
